### PR TITLE
[FIX] Make UserError 'You can not modify already invoiced timesheets …

### PR DESCRIPTION
**Pull request by CasVissers-360ERP**
[FIX] Make UserError 'You can not modify already invoiced timesheets (linked to a Sales order items invoiced on Time and material).' extendable

Description of the issue/feature this PR addresses:
The check prevents from changing timesheet lines which are already invoiced. This is a very valid check and applicable to a lot of customers. But not to all.

The use-case:
A task was related to a wrong sale order line and the user changes the sale order line on the task. The hours should move to this new sale order line and a credit note will be created from the old one.

Current behavior before PR:
Method for the check was not extendable, so no customization was possible.

Desired behavior after PR is merged:
Method for the check is extendable, so customization is possible.

opw-1974968

### CODE REVIEW

**Comment by CasVissers** [comment]
@len-odoo 
Any chance we can get this merged?

File: addons/sale_timesheet/models/account.py [hunk 1]
```diff
@@ -57,21 +57,15 @@ def _check_sale_line_in_project_map(self):
...
-        if 'task_id' in values and not values.get('so_line') and values.get('employee_id'):
-            task = self.env['project.task'].sudo().browse(values['task_id'])
-            employee = self.env['hr.employee'].sudo().browse(values['employee_id'])
-            values['so_line'] = self._timesheet_determine_sale_line(task, employee).id
-        return values
+    @api.multi
+    def _check_can_write(self, values):
+        if self.sudo().filtered(lambda aal: aal.so_line.product_id.invoice_policy == "delivery") and self.filtered(lambda timesheet: timesheet.timesheet_invoice_id):
+            if any([field_name in values for field_name in ['unit_amount', 'employee_id', 'project_id', 'task_id', 'so_line', 'amount', 'date']]):
+                raise UserError(_('You can not modify already invoiced timesheets (linked to a Sales order items invoiced on Time and material).'))
```
Comment (Yenthe666) [inline addons/sale_timesheet/models/account.py]: item instead of items?
Have you checked if the term is also in the `.pot` file by the way?

Comment (len-odoo) [reply addons/sale_timesheet/models/account.py]: You can see on 33092, where the diff is right, that the error message has not been modified.
This PR will be closed automatically when the other will be merged.

Comment (Yenthe666) [reply addons/sale_timesheet/models/account.py]: Ahh yes indeed. Thanks!

**Pull request closed without merging**